### PR TITLE
MKPC

### PR DESCRIPTION
**Pull request by trains2050**
i have made a presences for the popular website MKPC!!!!, if there any improvments that i have to make, please tell me.

### CODE REVIEW

**Comment by trains2050** [comment]
proof: 
image

File: websites/M/MKPC/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      "fr": "Vous connaissez certainement Mario Kart, le jeu de course le plus fun de tous les temps ! Mario Kart PC reprend les mêmes principes que le jeu original mais il est jouable sur navigateur, et gratuitement.  La plupart des modes issus de Mario Kart ont été repris : Grand Prix, courses VS, batailles de ballons, contre-la-montre... Et un dernier mode inédit : l'éditeur de circuits ! Placez les lignes droites et les virages, ajoutez les objets, insérez des accélérateurs... Tout est personnalisable ! Votre imagination est la seule limite ! Vous pouvez également partager vos créations et essayer celles des autres grâce à l'outil de partage. Plusieurs milliers de circuits ont déjà été partagés !  Enfin, il est possible d'affronter les joueurs du monde entier grâce au mode multijoueurs en ligne ! Grimpez dans le classement et devenez champion du monde !"
+    },
+    "service": "MKPC",
+    "version": "1.0.0",
+    "logo": "https://mkpc.malahieude.net/images/mkpc_box.jpg",
+    "thumbnail": "https://mkpc.malahieude.net/images/header_transparent2.png",
```
Comment (ririxidev) [inline websites/M/MKPC/dist/metadata.json:15]: imgur links only

File: websites/M/MKPC/presence.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  clientId: "812176837748457483"
+});
+
+var browsingStamp = Math.floor(Date.now() / 1000);
+
+var user: any;
+var title: any;
+var replace: any;
+var search: any;
+var post: any;
```
Comment (ririxidev) [inline websites/M/MKPC/presence.ts:11]: no vars, no any type.

File: websites/M/MKPC/presence.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "Viewing home page";
+  } else if (document.location.pathname.includes("topic.php?topic")) {
+    presenceData.startTimestamp = browsingStamp;
+   presenceData.details = "Browsing Forum topic";
+   presenceData.smallImageKey = "search"
+  } else if (document.location.pathname.includes("profil.php?")) {    // not spelling mistake, the document location name is profil not profile. 
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "Viewing user profile";
+    presenceData.smallImageKey = "search";
```
Comment (ririxidev) [inline websites/M/MKPC/presence.ts:36]: I'm 100% sure that you can get topic title or nickname on profile and if it is possible (100% is), add privacy setting

File: websites/M/MKPC/presence.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "browsing official topics ";
+    presenceData.smallImageKey = "search";
+  } else if (document.location.pathname == "/category.php?category=1") { 
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "Browsing unofficial announcements and presentations";
+    presenceData.smallImageKey = "search";
+  } else if (document.location.pathname == "/category.php?category=2") { 
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "Browsing discussion about the game ";
```
Comment (ririxidev) [inline websites/M/MKPC/presence.ts:51]: There shouldn't be space, or should it?

File: websites/M/MKPC/presence.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    presenceData.smallImageKey = "search";
+  } else if (document.location.pathname == "/category.php?category=3") { 
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "In Various discussions ";
```
Comment (ririxidev) [inline websites/M/MKPC/presence.ts:55]: There shouldn't be space, or should it?

File: websites/M/MKPC/presence.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    presenceData.smallImageKey = "search";
+  } else if (document.location.pathname == "/mariokart.php") {
+    presenceData.startTimestamp = browsingStamp;
+    presenceData.details = "Playing the core game";
+    presenceData.smallImageKey = "wheel"
```
Comment (ririxidev) [inline websites/M/MKPC/presence.ts:60]: You could get some info from this game, lap etc.

**Review by ririxidev — CHANGES_REQUESTED** [review]
You should look at these things:

File: websites/M/MKPC/presence.ts (hunk 2 above)
Comment (trains2050) [reply websites/M/MKPC/presence.ts:11]: Yes I know, will fix it XD, was planning to do var but website said no

File: websites/M/MKPC/presence.ts (hunk 6 above)
Comment (trains2050) [reply websites/M/MKPC/presence.ts:60]: Will try but Tbh I am limited with 2012 website

File: websites/M/MKPC/presence.ts (hunk 3 above)
Comment (trains2050) [reply websites/M/MKPC/presence.ts:36]: 🤣 i tried it using the html class and not possible, but sent a message to the website dev :)

Comment (trains2050) [reply websites/M/MKPC/presence.ts:36]: actually found out how

**Pull request closed without merging**